src/utils.zig

```diff
@@ -999,6 +1005,23 @@ pub const ThreadPool = struct {
 		isStillNeeded: *const fn(*anyopaque, milliTime: i64) bool,
 		run: *const fn(*anyopaque) void,
 		clean: *const fn(*anyopaque) void,
+		taskType: TaskType = .misc,
+	};
+	pub const Performance = struct {
+		mutex: std.Thread.Mutex = .{},
```

Comment:
I think this would be a great place to use atomics with monotonic ordering instead of locking.